src/v/datalake/schemaless_translator.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "bytes/iobuf.h"
+#include "iceberg/datatypes.h"
+
+#include <cstdint>
+#include <utility>
+
+namespace datalake {
+// We won't use this long-term. This is in place for development until the
+// iceberg_data PR is merged.
+struct schemaless_event_data {
+    iobuf key;
+    iobuf value;
+    int64_t timestamp;
+    int64_t offset;
+};
+
+class schemaless_translator {
+public:
+    schemaless_event_data translate_event(
+      iobuf&& key, iobuf&& value, int64_t timestamp, int64_t offset);
+
+    iceberg::struct_type get_schema();
```

Comment:
nit: these can both be const. Or will that not be the case in the future? Will other translator implementations need to fetch and cache a schema or something?

Reply:
They should have the schema.